src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,133 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.models.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sqlCommand = "INSERT INTO books (title,price) VALUE (?,?)";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(sqlCommand, Statement.RETURN_GENERATED_KEYS)) {
+
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("None of the rows have been added");
+            }
+
+            try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    long id = generatedKeys.getObject(1, Long.class);
+                    book.setId(id);
+                    return book;
+                } else {
+                    throw new DataProcessingException("Failed to retrieve the generated key");
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot create a connection to the DB", e);
```

Comment:
Let's change messages to describe more informative the issue like: 

Can't add a book: 
Can't find book by id ....

Reply:
Thank you for comment but I little bit disagree with you because in this case we make try catch for Connection field and if we catch some exception this exception will related to Connection field which is responsible for connection to the database.
Am I right?

Reply:
SQLException is related not only to Connection, but to PreparedStatement also. 
It's better to use try-catch for `public static Connection getConnection()` and provide informative message about connection issue, and in BookDaoImpl catch provide information about 
Can't add a book:
Can't find book by id ....